src/yarn.ts

```diff
@@ -45,6 +46,18 @@ export class Yarn {
   }
 
   private async findYarn(): Promise<string> {
-    return require.resolve('yarn/bin/yarn.js', {paths: [this.config.root, fileURLToPath(import.meta.url)]})
+    if (this.bin) return this.bin
+    try {
+      this.bin = require.resolve('yarn/bin/yarn.js', {paths: [this.config.root, fileURLToPath(import.meta.url)]})
```

Comment:
Why the different resolve logic for yarn? When I remove `yarn` from my linked `plugin-plugins` it resolves the to yarn that is included with the `sf` cli rather than falling back to my locally installed version
`@oclif/plugin-plugins:yarn yarn binary path /Users/ewillhoit/.nvm/versions/node/v20.12.2/lib/node_modules/@salesforce/cli/node_modules/yarn/bin/yarn.j`

Reply:
I just pulled the logic in exactly as it was in the last major: https://github.com/oclif/plugin-plugins/blob/4.3.10/src/yarn.ts#L30

I can replicate the same logic that we use for npm... but I'm wondering if it's worth it if we one day hope to remove the "install dependencies after linking" logic altogether. What do you think?

Reply:
I think that's fine since we'd like to get rid of this. I don't really see an issue with this, I was just curious if it was deliberate.